app/controllers/interactions_controller.rb

```diff
@@ -0,0 +1,9 @@
+class InteractionsController < ApplicationController
+  def index
+    # We parameterized this label to make it look like a slug for
+    # the URL, so now we need to turn it back into a string so that
+    # the service can be found from it. There's probably a better way.
+    @service = Service.find_by_label(params[:service_label].downcase.gsub("-"," "))
```

Comment:
OK. Adding a slug field wouldn't be too bad, and would make URLs look nicer for users, so I'll work on that.